src/core/hle/service/am/am.cpp

```diff
@@ -1759,6 +1779,57 @@ void ILibraryAppletSelfAccessor::PushInShowMiiEditData() {
     queue_data.emplace_back(std::move(argument_data));
 }
 
+void ILibraryAppletSelfAccessor::PushInShowSoftwareKeyboard() {
+    const Applets::CommonArguments arguments{
+        .arguments_version = Applets::CommonArgumentVersion::Version3,
+        .size = Applets::CommonArgumentSize::Version3,
+        .library_version = static_cast<u32>(Applets::SwkbdAppletVersion::Version524301),
+        .theme_color = Applets::ThemeColor::BasicBlack,
+        .play_startup_sound = true,
+        .system_tick = system.CoreTiming().GetClockTicks(),
+    };
+
+    const Applets::SwkbdConfigCommon swkbd_config{
+        .type = Applets::SwkbdType::Qwerty,
+        .ok_text{},
+        .left_optional_symbol_key{},
+        .right_optional_symbol_key{},
+        .use_prediction = false,
+        .key_disable_flags{},
+        .initial_cursor_position = Applets::SwkbdInitialCursorPosition::Start,
+        .header_text{},
+        .sub_text{},
+        .guide_text{},
+        .max_text_length = 500,
+        .min_text_length = 0,
+        .password_mode = Applets::SwkbdPasswordMode::Disabled,
+        .text_draw_type = Applets::SwkbdTextDrawType::Box,
+        .enable_return_button = true,
+        .use_utf8 = false,
+        .use_blur_background = true,
+        .initial_string_offset{},
+        .initial_string_length{},
+        .user_dictionary_offset{},
+        .user_dictionary_entries{},
+        .use_text_check = false,
+    };
+
+    Applets::SwkbdConfigNew swkbd_config_new{};
+
+    std::vector<u8> argument_data(sizeof(arguments));
+    std::vector<u8> swkbd_data(sizeof(swkbd_config));
+    std::vector<u8> swkbd_new_data(sizeof(swkbd_config_new));
+
+    std::memcpy(argument_data.data(), &arguments, sizeof(arguments));
+    std::memcpy(swkbd_data.data(), &swkbd_config, sizeof(swkbd_config));
+    std::memcpy(swkbd_new_data.data(), &swkbd_config_new, sizeof(Applets::SwkbdConfigNew));
+    std::memcpy(&swkbd_config_new, swkbd_data.data() + sizeof(Applets::SwkbdConfigCommon),
```

Comment:
Again this should be a work buffer
```suggestion
    std::memcpy(&swkbd_work_buffer, &initial_string, swkbd_config.initial_string_length * sizeof(char16_t));
```